Packs/AzureWAF/Integrations/AzureWAF/AzureWAF.py

```diff
@@ -0,0 +1,375 @@
+            method='GET',
+            url_suffix=f'/resourceGroups/{resource_group_name}/providers/'
+                       f'Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies'
+        )
+
+    def get_policy_list_by_subscription_id(self) -> Dict:
+        return self.http_request(
+            method='GET',
+            url_suffix='/providers/Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies'
+        )
+
+    def update_policy_upsert(self, policy_name: str, resource_group_name: str, data: Dict) -> Dict:
+        return self.http_request(
+            method='PUT',
+            url_suffix=f'/resourceGroups/{resource_group_name}/providers/'
+                       f'Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies/{policy_name}',
+            data=data
+        )
+
+    def delete_policy(self, policy_name: str, resource_group_name: str):
+        return self.http_request(
+            method='DELETE',
+            return_empty_response=True,
+            resp_type='response',
+            url_suffix=f'/resourceGroups/{resource_group_name}/providers/'
+                       f'Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies/{policy_name}',
+        )
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_connection(client: AzureWAFClient, params: Dict):
+    if params.get('self_deployed', False) and not params.get('auth_code'):
+        return_error('You must enter an authorization code in a self-deployed configuration.')
+    client.ms_client.get_access_token()  # If fails, MicrosoftApiModule returns an error
+    return '✅ Great Success!'
+
+
+@logger
+def start_auth(client: AzureWAFClient) -> CommandResults:
+    user_code = client.ms_client.device_auth_request()
+    return CommandResults(readable_output=f"""### Authorization instructions
+        1. To sign in, use a web browser to open the page:
+            [https://microsoft.com/devicelogin](https://microsoft.com/devicelogin)
+           and enter the code **{user_code}** to authenticate.
+        2. Run the **!azure-waf-auth-complete** command in the War Room.""")
+
+
+@logger
+def complete_auth(client: AzureWAFClient):
+    client.ms_client.get_access_token()
+    return '✅ Authorization completed successfully.'
+
+
+def policy_get_command(client: AzureWAFClient, **args) -> CommandResults:
+    """
+    Gets resource group name (or taking instance's default one) and policy name(optional).
+    If a policy name provided, Retrieve the policy by name and resource group.
+    Otherwise, retrieves all policies within the resource group.
+    """
+
+    policy_name: str = args.get('policy_name', '')
+    resource_group_name: str = args.get('resource_group_name', client.resource_group_name)
+    verbose = True if args.get("verbose", "false") == "true" else False
+    limit = str(args.get("limit"))
+    if not limit.isdigit():
+        raise Exception("please provide a numeric limit")
+    limit = int(limit)
```

Comment:
why would it happen? and let's add a default. e.g:
```suggestion
    limit = int(args.get("limit", '20'))
```